protocol/src/main/java/org/geysermc/mcprotocollib/auth/util/UUIDUtils.java

```diff
@@ -0,0 +1,26 @@
+package org.geysermc.mcprotocollib.auth.util;
+
+import java.util.UUID;
+
+public class UUIDUtils {
+    public static UUID convertToDashed(String noDashes) {
+        if (noDashes == null) {
+            return null;
+        }
+
+        StringBuilder idBuff = new StringBuilder(noDashes);
+        idBuff.insert(20, '-');
+        idBuff.insert(16, '-');
+        idBuff.insert(12, '-');
+        idBuff.insert(8, '-');
+        return UUID.fromString(idBuff.toString());
```

Comment:
This seems simpler:
```java
return new UUID(
    HexFormat.fromHexDigitsToLong(noDashes, 0, 16),
    HexFormat.fromHexDigitsToLong(noDashes, 16, 32)
);
```

Reply:
That seems slower to me

Reply:
I'll do a microbenchmark

Reply:
![image](https://github.com/GeyserMC/MCProtocolLib/assets/40795980/22c41bac-3622-4f68-9e67-7d56fc708986)
Alright seems like your code is 3.6x faster than mine.